### ISSUE TITLE
Ingress controller pod is not starting successfully

### ISSUE THREAD

**Issue by srinisachanta** [Azure/application-gateway-kubernetes-ingress#806]
**Describe the bug**
Ingress Controller Pod is not starting. I am using AKS Cluster Provisioned using managed identity (currently in preview)

**To Reproduce**
Steps to reproduce the behavior: Followed brownfield deployment guide.

**Ingress Controller details**
**k describe pod appgw-ingress-azure**
Type     Reason     Age                   From                                         Message
  ----     ------     ----                  ----                                         -------
  Normal   Scheduled  11m                   default-scheduler                            Successfully assigned default/appgw-ingress-azure-6cc9846c47-s88np to aks-aksdevpool-56626102-vmss000000
  Normal   Pulling    10m (x2 over 11m)     kubelet, aks-aksdevpool-56626102-vmss000000  Pulling image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.0.0"
  Normal   Pulled     10m (x2 over 11m)     kubelet, aks-aksdevpool-56626102-vmss000000  Successfully pulled image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.0.0"
  Normal   Created    10m (x2 over 11m)     kubelet, aks-aksdevpool-56626102-vmss000000  Created container ingress-azure 
  Normal   Started    10m (x2 over 11m)     kubelet, aks-aksdevpool-56626102-vmss000000  Started container ingress-azure 
  Normal   Killing    10m                   kubelet, aks-aksdevpool-56626102-vmss000000  Container ingress-azure failed liveness probe, will be restarted
  Warning  Unhealthy  9m29s (x5 over 10m)   kubelet, aks-aksdevpool-56626102-vmss000000  Liveness probe failed: Get http://10.0.0.20:8123/health/alive: dial tcp 10.0.0.20:8123: connect: connection refused
  Warning  Unhealthy  6m20s (x30 over 11m)  kubelet, aks-aksdevpool-56626102-vmss000000  Readiness probe failed: Get http://10.0.0.20:8123/health/ready: dial tcp 10.0.0.20:8123: connect: connection refused
  Warning  BackOff    75s (x16 over 5m7s)   kubelet, aks-aksdevpool-56626102-vmss000000  Back-off restarting failed container

**k logs pod appgw-ingress-azure-6cc9846c47-s88np**
Error from server (NotFound): pods "pod" not found
PS C:\gitrepos\AKSPlayGround> k logs appgw-ingress-azure-6cc9846c47-s88np    
ERROR: logging before flag.Parse: I0413 00:50:38.830051       1 main.go:302] Using verbosity level 3 from environment variable APPGW_VERBOSITY_LEVEL
I0413 00:50:38.867838       1 environment.go:168] KUBERNETES_WATCHNAMESPACE is not set. Watching all available namespaces.
I0413 00:50:38.867861       1 main.go:132] App Gateway Details: Subscription: 00000000-0000-0000-0000-000000000000, Resource Group: Admin-CentralUS-RG, Name: AppGateway-CentralUS
I0413 00:50:38.867868       1 auth.go:90] Creating authorizer from Azure Managed Service Identity

Thanks for any inputs

**Comment by akshaysngupta** [comment]
@srinisachanta Can you check the logs for AAD Pod identity ? Look for nmi and mic pod logs for clue on why AGIC pod is not able to get a token.

**Comment by srinisachanta** [comment]
@akshaysngupta, you are spot on. MSI related to agent pool was missing write permission on VMSS. Not sure why, I manually gave access and redeployed and errors went away and ingress deployment was successful. Just now I removed role assignments I granted and redeployed  but things are still working so not quite sure whats happening and don't see any errors. Anyways if I find any issues I will log an issue. Thanks for your help!
